### PR TITLE
Update opentelemetry operations collector to otel 0.39

### PR DESCRIPTION
**Pull request by schmikei**
This adds a new version of the metadata-gen cmd and also refactors to the latest standards by the `opentelemetry-collector`

### CODE REVIEW

File: receiver/couchdbreceiver/metadata.yaml [hunk 1]
```diff
@@ -16,78 +16,78 @@ metrics:
...
   couchdb.writes:
     description: The database write count.
     unit: 1
     data:
       type: sum
       monotonic: true
       aggregation: cumulative
-    labels: [ node_name ]
\ No newline at end of file
+    attributes: [ node_name ]
```
Comment (Mrod1598) [inline receiver/couchdbreceiver/metadata.yaml:93]: I know there wasn't a new line before but could you add one? They always say new lines at the end of `yaml` files and `json`

File: receiver/couchdbreceiver/scraper.go [hunk 2]
```diff
@@ -64,20 +64,20 @@ func addToIntMetric(metric pdata.NumberDataPointSlice, attributes pdata.Attribut
...
-		return pdata.ResourceMetricsSlice{}, errors.New("failed to fetch couchdb stats")
+		return pdata.Metrics{}, errors.New("failed to fetch couchdb stats")
 	}
 
 	// metric initialization
-	rms := pdata.NewResourceMetricsSlice()
-	ilm := rms.AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty()
+	rms := pdata.NewMetrics()
+	ilm := rms.ResourceMetrics().AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty()
 	ilm.InstrumentationLibrary().SetName("otel/postgresql")
```
Comment (Mrod1598) [inline receiver/couchdbreceiver/scraper.go:81]: Dan pointed out to me last time that apparently this needs to be updated to `otelcol/` for the prefix for scrapers now.

File: receiver/elasticsearchreceiver/metadata.yaml [hunk 3]
```diff
@@ -54,156 +54,156 @@ metrics:
...
+    attributes: [thread_pool_name]
   elasticsearch.thread_pool.completed:
     description: Number of tasks completed by the thread pool executor.
     unit: 1
     data:
       type: sum
       aggregation: cumulative
-    labels: [thread_pool_name]
\ No newline at end of file
+    attributes: [thread_pool_name]
```
Comment (Mrod1598) [inline receiver/elasticsearchreceiver/metadata.yaml:209]: new line here as well.

File: receiver/mysqlreceiver/metadata.yaml [hunk 4]
```diff
@@ -64,104 +64,104 @@ metrics:
...
-    labels: [ sorts]
+    attributes: [ sorts]
   mysql.threads:
     description: Thread count
     unit: 1
     data:
       type: gauge
-    labels: [ threads]
\ No newline at end of file
+    attributes: [ threads]
```
Comment (Mrod1598) [inline receiver/mysqlreceiver/metadata.yaml:167]: newline

**Review by Mrod1598 — COMMENTED** [review]
Looks Good just a couple small things otel authors are always hitting us on.

File: receiver/couchdbreceiver/scraper.go (hunk 2 above)
Comment (schmikei) [reply receiver/couchdbreceiver/scraper.go:81]: Wait could you elaborate a tad bit more on what needs to change here?

**Review by Mrod1598 — APPROVED** [review]

**Pull request merged**